setup.py

```diff
@@ -129,40 +133,81 @@ def run(self):
         SDist.run(self)
 
 
-with open('requirements.txt') as requirements_file:
-    install_requirements = requirements_file.read().splitlines()
-    if not install_requirements:
-        print("Unable to read requirements from the requirements.txt file"
-              "That indicates this copy of the source code is incomplete.")
+def read_file(file_name):
+    """Read file and return its contents."""
+    with open(file_name, 'r') as f:
+        return f.read()
+
+
+def read_requirements(file_name):
+    """Read requirements file as a list."""
+    reqs = read_file(file_name).splitlines()
+    if not reqs:
+        raise RuntimeError(
+            "Unable to read requirements from the %s file"
+            "That indicates this copy of the source code is incomplete."
+            % file_name
+        )
         sys.exit(2)
+    return reqs
 
-# pycrypto or cryptography.   We choose a default but allow the user to
-# override it.  This translates into pip install of the sdist deciding what
-# package to install and also the runtime dependencies that pkg_resources
-# knows about
-crypto_backend = os.environ.get('ANSIBLE_CRYPTO_BACKEND', None)
-if crypto_backend:
-    if crypto_backend.strip() == 'pycrypto':
-        # Attempt to set version requirements
-        crypto_backend = 'pycrypto >= 2.6'
 
-    install_requirements = [r for r in install_requirements if not (r.lower().startswith('pycrypto') or r.lower().startswith('cryptography'))]
-    install_requirements.append(crypto_backend)
+def append_crypto_to_req(req):
+    """Append custom ANSIBLE_CRYPTO_BACKEND to requirements list.
 
-# specify any extra requirements for installation
-extra_requirements = dict()
-extra_requirements_dir = 'packaging/requirements'
-for extra_requirements_filename in os.listdir(extra_requirements_dir):
-    filename_match = re.search(r'^requirements-(\w*).txt$', extra_requirements_filename)
-    if filename_match:
-        with open(os.path.join(extra_requirements_dir, extra_requirements_filename)) as extra_requirements_file:
-            extra_requirements[filename_match.group(1)] = extra_requirements_file.read().splitlines()
+    pycrypto or cryptography. We choose a default but allow the user to
+    override it. This translates into pip install of the sdist deciding what
+    package to install and also the runtime dependencies that pkg_resources
+    knows about.
+    """
+    crypto_backend = os.environ.get('ANSIBLE_CRYPTO_BACKEND', '').strip()
+    if not crypto_backend:
+        return req
+
+    PYCRYPTO_DIST = 'pycrypto'
+    if crypto_backend.strip() == PYCRYPTO_DIST:
+        # Attempt to set version requirements
+        crypto_backend = '%s >= 2.6' % PYCRYPTO_DIST
+
+    CRYPTO_LIBS = PYCRYPTO_DIST, 'cryptography'
+    is_crypto = lambda r: any(_.lower().startswith(c) for c in CRYPTO_LIBS)
+    is_not_crypto = lambda r: not is_crypto(r)
+
+    return list(filter(is_not_crypto, req)) + [crypto_backend]
```

Comment:
We should avoid lambdas and filters most of the time.  Use list comprehensions and generator expressions instead.  It's both faster and more pythonic to use those idioms.

Reply:
My motivation was to keep lines short and readable. FP isn't bad, but I hear you: I'll revert to comprehensions, but will keep lambda as a function for readability.